### PR TITLE
[ManifestAlloc] Handle TupleType inputs in CheckReshapeOnly

### PR DESCRIPTION
**Pull request by rohanmukh**
This arises insed ManifestAllocPass inside relay.vm.compile

Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by anijain2305** [comment]
@icemelon9 Can you PTAL?

**Comment by jroesch** [comment]
cc @altanh @MarisaKirisame

File: tests/python/relay/test_vm.py [hunk 1]
```diff
@@ -754,5 +754,21 @@ def test_vm_reshape_tensor():
...
+        type_annotation=relay.TupleType([relay.TensorType(x_shape), relay.TensorType(y_shape)]),
+    )
+    out = relay.reshape(relay.TupleGetItem(tup, 0), (1, -1))
+    f = relay.Function([tup], out)
+
+    x_data = np.random.uniform(size=x_shape).astype("float32")
+    y_data = np.random.uniform(size=y_shape).astype("float32")
+
+    for tgt, ctx in tvm.testing.enabled_targets():
+        res = veval(f, (x_data, y_data))
```
Comment (altanh) [inline tests/python/relay/test_vm.py:769]: use `tgt` and `ctx` here?

Comment (rohanmukh) [reply tests/python/relay/test_vm.py]: Thanks @altanh for pointing this. I resolved this.

**Review by icemelon9 — APPROVED** [review]
lgtm

**Comment by anijain2305** [comment]
Thanks @rohanmukh @altanh @icemelon9 
This is merged

**Pull request merged**